### PR TITLE
Add pet person trait to trinket category which spawns you with random pet

### PR DESCRIPTION
**Pull request by Glamurio**
[Traits] [Feature]

## About the PR 
Adds a new "Pet Person" trait that replaces your starting trinket with a carrier containing a random `/mob/living/critter/small_animal/` with the new `player_can_spawn_with_pet` boolean set to true.

- Changes `randomize_name` to `random_name` and `randomize_look` to `random_look`
- Moves `var/random_name` and `var/random_look` from `/mob/living/critter/small_animal/cat` to `/mob/living/critter/small_animal`.
- Creates `proc/randomize_name` in `/mob/living/critter/small_animal` to randomize names.
- Moves `var/aggressive` from `/mob/living/critter/small_animal/scorpion` and `/mob/living/critter/small_animal/rattlesnake` to `/mob/living/critter/small_animal`.

- Introduces a new `var/name_list` in `/mob/living/critter/small_animal` which is used to randomize names. It defaults to `names/pets.txt`.
- Creates `names/dogs.txt` and `names/pets.txt`.

Worth noting, there is a `is_pet` attribute that is currently used to track pets via `START_TRACKING_CAT(TR_CAT_PETS)`. The implementation seems arbitrary, it seems to mostly track command pets like "Jones", but then also randomly track all seals. For the time being, I've made it so that all player pets are also tracked via the same tracker, but if necessary we can separate this.

Icon:
image

## Why's this needed? 
Traits are cool, and ever since we've gotten pet carriers, I've been wanting to add more ways for people to have "pets". This is a pretty good way to implement something like this in a way that doesn't seem overbearing. I've ensured that all pets have a friendly AI and are harmless, but do retaliate once when attacked but they won't chase you down until death.

## Changelog 

```changelog
(u)Glamurio (Ryou) & Zac
(*)Added the "Pet Person" trait, which spawns you with a random pet.
```

### CODE REVIEW

File: code/mob/living/critter/small_animal.dm [hunk 1]
```diff
@@ -72,6 +72,12 @@ ABSTRACT_TYPE(/mob/living/critter/small_animal)
 	ai_retaliate_patience = 2
 	ai_retaliate_persistence = RETALIATE_ONCE
 
+	var/aggressive = FALSE
+	var/random_name = FALSE
+	var/random_look = FALSE
+	var/name_list = "names/pets.txt"
+	var/player_can_spawn_with_pet = FALSE // For the Pet Person trait that makes you spawn with a pet (do not confuse with is_pet!!!)
```
Comment (TobleroneSwordfish) [inline code/mob/living/critter/small_animal.dm:79]:
```suggestion
	var/player_can_spawn_with_pet = FALSE //! For the Pet Person trait that makes you spawn with a pet (do not confuse with is_pet!!!)
```
doc comment

File: code/mob/living/critter/small_animal.dm [hunk 2]
```diff
@@ -84,13 +90,15 @@ ABSTRACT_TYPE(/mob/living/critter/small_animal)
 			src.is_pet = (copytext(src.name, 1, 2) in uppercase_letters)
 		if(in_centcom(loc) || current_state >= GAME_STATE_PLAYING)
 			src.is_pet = 0
-		if(src.is_pet)
+		if(src.is_pet || src.player_can_spawn_with_pet) // Also track player pets
```
Comment (TobleroneSwordfish) [inline code/mob/living/critter/small_animal.dm:93]: Probably best to only track it if it *is* a pet, not if this type can spawn as one?

File: code/mob/living/critter/small_animal.dm (hunk 1 above)
Comment (TobleroneSwordfish) [inline code/mob/living/critter/small_animal.dm:79]: Also note to whoever merges this: will need secret changes to blacklist a few secret critters from appearing here.

File: code/mob/living/critter/small_animal.dm [hunk 3]
```diff
@@ -329,19 +350,20 @@ ABSTRACT_TYPE(/mob/living/critter/small_animal)
...
 	var/attack_damage = 3
 
 	New()
 		..()
-		if (src.randomize_name)
-			src.name = pick_string_autokey("names/cats.txt")
-			src.real_name = src.name
-		if (src.randomize_look)
+		if (src.random_name)
+			src.randomize_name()
```
Comment (TobleroneSwordfish) [inline code/mob/living/critter/small_animal.dm:365]: Shouldn't the parent do this now?

File: strings/names/dogs.txt [hunk 4]
```diff
@@ -0,0 +1,30 @@
+
+Sure, here's an extended list of dog names formatted similarly to the provided list of cat names:
```
Comment (TobleroneSwordfish) [inline strings/names/dogs.txt:2]: pfff probably best to remove the chatgpt comment here
```suggestion
```

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
I like this idea, very cute

File: strings/names/dogs.txt (hunk 4 above)
Comment (Glamurio) [reply strings/names/dogs.txt:2]: Bahaha that's funny my bad

File: code/mob/living/critter/small_animal.dm (hunk 2 above)
Comment (Glamurio) [reply code/mob/living/critter/small_animal.dm:93]: Mentioned this as well, what is considered a "pet" by the game is very arbitrary. It tracks random things like seals, but also command pets.

**Review by Tarmunora — CHANGES_REQUESTED** [review]
Should not count towards Noah's Shuttle medal

**Comment by Glamurio** [comment]
> Should not count towards Noah's Shuttle medal

Right. So that's what the tracker is for. That makes sense, I've removed them.

File: code/procs/jobprocs.dm [hunk 5]
```diff
@@ -704,6 +704,16 @@ else if (istype(JOB, /datum/job/security/security_officer))\
 			trinket = new/obj/item/reagent_containers/food/snacks/ingredient/egg/bee/buddy(src)
 		else
 			trinket = new/obj/item/reagent_containers/food/snacks/ingredient/egg/bee(src)
+	else if (src.traitHolder && src.traitHolder.hasTrait("petperson"))
+		var/obj/item/pet_carrier/carrier = new/obj/item/pet_carrier(src)
+		var/picked = pick(filtered_concrete_typesof(/mob/living/critter/small_animal/, /proc/filter_carrier_pets))
```
Comment (Tarmunora) [inline code/procs/jobprocs.dm:709]:
```suggestion
		var/picked = pick(filtered_concrete_typesof(/mob/living/critter/small_animal/, GLOBAL_PROC_REF(filter_carrier_pets)))
```

File: code/procs/jobprocs.dm [hunk 6]
```diff
@@ -704,6 +704,16 @@ else if (istype(JOB, /datum/job/security/security_officer))\
...
+	else if (src.traitHolder && src.traitHolder.hasTrait("petperson"))
+		var/obj/item/pet_carrier/carrier = new/obj/item/pet_carrier(src)
+		var/picked = pick(filtered_concrete_typesof(/mob/living/critter/small_animal/, GLOBAL_PROC_REF(filter_carrier_pets)))
+		var/mob/living/critter/small_animal/pet = new picked(src)
+		pet.ai_type = /datum/aiHolder/wanderer
+		pet.aggressive = FALSE
+		pet.randomize_name()
+		pet.ai_retaliate_persistence = RETALIATE_ONCE
+		carrier.trap_mob(pet, src)
+		src.force_equip(carrier, SLOT_IN_BACKPACK, TRUE)
```
Comment (TobleroneSwordfish) [inline code/procs/jobprocs.dm:716]: Pet carriers don't actually fit in backpacks so it's a little strange that it starts there, also some jobs (clown) don't start with a backpack. Probably better to just go `l_hand` -> `r_hand` -> floor, I don't think there are any jobs that spawn with two items in hand, but there might be other trait combinations.

File: code/datums/traits.dm [hunk 7]
```diff
@@ -483,6 +483,13 @@
 	points = -1
 	category = list("trinkets")
 
+/datum/trait/petperson
```
Comment (TobleroneSwordfish) [inline code/datums/traits.dm:486]: It would be really nice (but not essential) if this had an icon for the trait screen.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Quickly tested, looks mostly good just needs some equip logic fixing.

File: code/procs/jobprocs.dm (hunk 6 above)
Comment (Glamurio) [reply code/procs/jobprocs.dm:716]: Oh that's a goof, I was so sure they fit in backpacks

File: code/datums/traits.dm (hunk 7 above)
Comment (Glamurio) [reply code/datums/traits.dm:486]: Yeah that would be sick, I asked some peeps, will see if something turns up, any sprite I would whip up would be mediocre

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**